frontend/views/containers/contributions/TodoHistory.vue

```diff
@@ -38,45 +37,42 @@ export default ({
   computed: {
     ...mapGetters([
       'currentPaymentPeriod',
-      'groupCreatedDate'
+      'groupCreatedDate',
+      'thisPeriodPaymentInfo'
     ])
   },
   created () {
     this.updateHistory()
   },
   methods: {
     async updateHistory () {
-      this.history = []
-
-      let period = null
-      const firstDistributionPeriod = await this.historicalPeriodStampGivenDate(this.groupCreatedDate)
       const getLen = obj => Object.keys(obj).length
-
-      for (let i = 0; i < MAX_HISTORY_PERIODS; i++) {
-        period = period === null ? this.currentPaymentPeriod : await this.historicalPeriodBeforePeriod(period)
-        if (!period || comparePeriodStamps(period, firstDistributionPeriod) < 0) break
-
+      const allPeriods = await this.getAllSortedPeriodKeys()
+      const periods = allPeriods.slice(-MAX_HISTORY_PERIODS)
+      this.history = await Promise.all(periods.map(async (period) => {
         const paymentDetails = await this.getPaymentDetailsByPeriod(period)
         const { lastAdjustedDistribution } = await this.getPaymentPeriod(period)
-
         const doneCount = getLen(paymentDetails)
         const markedAsNotReceivedCount = Object.values(paymentDetails)
           .filter(({ data }) => data.status === PAYMENT_NOT_RECEIVED).length
         const missedCount = getLen(lastAdjustedDistribution || {})
-        this.history.unshift({
+        return {
           total: doneCount === 0 ? 0 : (doneCount - markedAsNotReceivedCount) / (doneCount + missedCount),
-          title: this.getPeriodFromStartToDueDate(period),
+          title: this.getPeriodFromStartToDueDate(period, periods),
           tooltipContent: [
             L('Total: {total}', { total: doneCount + missedCount }),
             L('Completed: {completed}', { completed: doneCount - markedAsNotReceivedCount })
           ]
-        })
-      }
+        }
+      }))
     }
   },
   watch: {
     currentPaymentPeriod () {
       this.updateHistory()
+    },
+    thisPeriodPaymentInfo () {
+      this.updateHistory()
```

Comment:
Before the graphs wouldn't update when a payment was made, now they do because of this code